VirtoCommerce.Storefront.Model/Catalog/Product.cs

```diff
@@ -365,12 +365,16 @@ private void ApplyPricesWithExchangeRates(IEnumerable<Currency> allCurrencies)
                 if (price == null)
                 {
                     price = new ProductPrice(currency);
+
                     //Convert exist price to new currency
                     if (Prices.Any())
                     {
                         price = Prices.First().ConvertTo(currency);
                         price.TierPrices.Add(new TierPrice(price.SalePrice, 1));
                     }
+
+                    // Product ID must exist in product price. TODO: Add it to product price constructor
```

Comment:
Why you don't added id to the Product price constructor now?

Reply:
Because it coming from vc-storefront (base), require rewiring a code in a lot of places and will cause our fork incompatible with original storefront. My propoposal is to do it in vc-storefront and then just merge to our